### PR TITLE
Upgrade to KataGo v1.12.4

### PR DESCRIPTION
**Pull request by tomtseng**
Update `KataGo-custom` to v1.12.4.

Model training still uses TF so that we can finetune our TF cyclic adversary. If we want to finetune the new architectures  using PyTorch model training, we'll need to further edit kubernetes/train.sh since some of the flags to train.py have changed (e.g., model kind is a flag -model-kind instead of a positional argument, and warmstarting is done with the -initial-checkpoint flag instead of done by copying files into a initial_weights directory). `kubernetes/shuffle_and_export.sh` would need to switch to using the PyTorch code as well, since the `shuffleddata` format is different between TF and PyTorch.

### CODE REVIEW

File: configs/gtp-amcts.cfg [hunk 1]
```diff
@@ -8,9 +8,19 @@ numBots = 2
...
 # This is set to 6 in the original gtp_example.cfg but A-MCTS only supports numSearchThreads = 1
 numSearchThreads = 1
+
+# If we want to keep behavior close to KataGo v1.10, consider setting the
+# following parameters:
+# valueWeightExponent0 = 0.5
+# subtreeValueBiasFactor0 = 0.35
+# subtreeValueBiasWeightExponent0 = 0.8
+# fpuParentWeightByVisitedPolicy0 = false
+# enablePassingHacks0 = false
```
Comment (tomtseng) [inline configs/gtp-amcts.cfg:26]: I think `fpuParentWeightByVisitedPolicy0` is the important one, not sure if the other ones affect win rate much

File: configs/gtp-amcts.cfg [hunk 2]
```diff
@@ -17,10 +17,14 @@ useGraphSearch1 = true
...
-# valueWeightExponent0 = 0.5
-# subtreeValueBiasFactor0 = 0.35
-# subtreeValueBiasWeightExponent0 = 0.8
-# fpuParentWeightByVisitedPolicy0 = false
-# enablePassingHacks0 = false
+# valueWeightExponent = 0.5
+# subtreeValueBiasFactor = 0.35
+# subtreeValueBiasWeightExponent = 0.8
+# fpuParentWeightByVisitedPolicy = false
+# enablePassingHacks = false
```
Comment (tomtseng) [inline configs/gtp-amcts.cfg:30]: fixing these params slightly — want to tweak these params not just for the adv but also for the adv's model of the victim. this actually does affect behavior, I think was seeing worse GTP win rates otherwise

File: .gitmodules [hunk 3]
```diff
@@ -7,3 +7,6 @@
 [submodule "engines/KataGo-raw"]
 	path = engines/KataGo-raw
 	url = git@github.com:lightvector/KataGo.git
+[submodule "engines/KataGo-tensorflow"]
```
Comment (AdamGleave) [inline .gitmodules:10]: To check I understand: this is an older version of KataGo-custom before training switched to PyTorch?

There's no way of doing TensorFlow training with later versions of KataGo-custom?

Having three versions to keep track of is going to be tricky but if it's unavoidable we can manage it.

File: configs/gtp-amcts.cfg [hunk 4]
```diff
@@ -17,6 +17,10 @@ useGraphSearch1 = true
 # This is set to 6 in the original gtp_example.cfg but A-MCTS only supports numSearchThreads = 1
 numSearchThreads = 1
 
+# conservativePass is "a good idea when using [the bot] for analysis or playing
```
Comment (AdamGleave) [inline configs/gtp-amcts.cfg:20]: Can you remind me what the purpose of this file is? Are we trying to match settings to what we use during AMCTS victim-play? Is there a concrete use case for this? (Trying to work out if this flag makes sense given that use case.)

**Review by AdamGleave — APPROVED** [review]
LGTM -- couple of minor clarifying questions, you may want me to re-review if those questions give rise to any reconsiderations, but otherwise feel free to go ahead and merge.

File: configs/gtp-amcts.cfg (hunk 4 above)
Comment (tomtseng) [reply configs/gtp-amcts.cfg:20]: We use this for GTP evals and transfer experiments, and in these experiments we're always using Tromp-Taylor

File: .gitmodules (hunk 3 above)
Comment (tomtseng) [reply .gitmodules:10]: yeah, the tensorflow training/shuffling code has all been removed in KataGo v1.12. We need to either have this extra `engines/KataGo-tensorflow` submodule, or copy all the old tensorflow code back into `KataGo-custom`
